server/src/domain/asset/asset.service.ts

```diff
@@ -420,11 +436,82 @@ export class AssetService {
       await this.assetRepository.updateAll([options.stackParentId], { stackParentId: null });
     }
 
+    if (creationDate !== undefined || latitude !== undefined || longitude !== undefined) {
+      for (const id of ids) {
+        const { pathMetadata, newExifData } = await this.getPathAndExif(id, {
+          CreationDate: creationDate,
+          GPSLatitude: latitude,
+          GPSLongitude: longitude,
+        });
+        await this.assetRepository.save({ id, sidecarPath: pathMetadata });
+        await this.assetRepository.writeExif(pathMetadata, newExifData);
+        await this.jobRepository.queue({ name: JobName.METADATA_EXTRACTION, data: { id: id } });
+      }
+    }
+
```

Comment:
Should this (and the section from `update()`) be extracted to a separate shared function?

Reply:
Yeah you are right it's now factorized